File: C99/백준/Bronze/9506. 약수들의 합/약수들의 합.c
#include<stdio.h>
int main(void)
{
	int n, sum;
	while(1)
	{
		scanf("%d", &n);
		if (n == -1)
		{
			break;
		}
		sum = 0;
		for (int i = 1; i <= n / 2; i++)
		{
			if (n % i == 0)
			{
				sum += i;
			}
		}
		if (sum == n)
		{
			printf("%d = ", n);
			for (int i = 1; i <= n / 2; i++)
			{
				if (n % i == 0 && i <= n / 2 - 1)
				{
					printf("%d + ", i);
				}
				if(n % i == 0 && i > n / 2 - 1)
				{
					printf("%d\n", i);
				}
			}
		}
		else
		{
			printf("%d is NOT perfect.\n", n);
		}
	}
	
	return 0;
}

File: Python/백준/Bronze/9506. 약수들의 합/약수들의 합.py
import sys

def divisor(n):
    result = []
    for i in range(1, int(n**(1/2))+1):
        if(n%i==0):
            result.append(i)
            if(i**2!=n):
                result.append(n//i)
    result.sort()
    return result

while(1):
    n = int(sys.stdin.readline().strip())
    if(n==-1):
        break
    
    result = divisor(n)
    sum = 0
    for i in result[:-1]:
        sum += i
    
    print(n,end="")    
    if(sum == n):
        print(" = ", end = "")
        for i in result:
            if(i < result[len(result) - 2]):
                print("%d + "%i, end = "")
            elif(i == result[-2]):
                print("%d"%i)
    else:
        print(" is NOT perfect.")